File: articles/counterfactual-reasoning-in-algorithmic-systems/haskell/CounterfactualReview.hs
module Main where

data Decision = Favorable | NotFavorable deriving (Eq, Show)

label :: Double -> Double -> Decision
label threshold score = if score >= threshold then Favorable else NotFavorable

flipped :: Double -> Double -> Double -> Bool
flipped threshold original counterfactual = label threshold original /= label threshold counterfactual

main :: IO ()
main = print (flipped 0.62 0.57 0.65)

File: articles/counterfactual-reasoning-in-algorithmic-systems/go/counterfactual_review.go
package main

import "fmt"

func label(score float64, threshold float64) string {
	if score >= threshold {
		return "favorable"
	}
	return "not_favorable"
}

func main() {
	original := label(0.57, 0.62)
	counterfactual := label(0.65, 0.62)
	fmt.Printf("original_label=%s\n", original)
	fmt.Printf("counterfactual_label=%s\n", counterfactual)
	fmt.Printf("decision_flipped=%t\n", original != counterfactual)
}
